content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,615 @@
+
+### Source Code
+
+You can find the complete source code for all of this tutorial's examples in the [code samples section of this website's repository]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/).
+
+## Rationale
+
+This tutorial takes you through the process of creating a XRP Wallet application from scratch. Starting with a simple,
+"Hello World" like example with minimal functionality, step-by-step we will add more and more complex features.
+
+As JavaScript is a programming language that originated in the web browser ecosystem, it's not natively 
+supporting the creation of desktop applications. We have to pick a frameworks that enable us to write 
+desktop applications with JavaScript. For this tutorial we will use the [Electron Framework](https://www.electronjs.org/), as it is well 
+established, documented and will get us up and running without having to care for dependencies and stuff that would make 
+us divert too much from the goals of this tutorial.
+
+## Goals
+
+At the end of this tutorial, you will have built a JavaScript Wallet application that looks something like this:
+
+![Desktop wallet screenshot](img/javascript-wallet-preview.png)
+
+The look and feel of the user interface should be roughly the same regardless of operating system, as the Electron Framework allows us to write 
+cross-platform applications that are styled with HTML and CSS just like a web-based application.
+
+The application we are going to build here will be capable of the following:
+
+- Showing updates to the XRP Ledger in real-time.
+- Viewing any XRP Ledger account's activity "read-only" including showing how much XRP was delivered by each transaction.
+- Showing how much XRP is set aside for the account's [reserve requirement](reserves.html).
+- Sending [direct XRP payments](direct-xrp-payments.html), and providing feedback about the intended destination address, including:
+    - Whether the intended destination already exists in the XRP Ledger, or the payment would have to fund its creation.
+    - If the address doesn't want to receive XRP ([`DisallowXRP` flag](become-an-xrp-ledger-gateway.html#disallow-xrp) enabled).
+    - If the address has a [verified domain name](https://xrpl.org/xrp-ledger-toml.html#account-verification) associated with it.
+
+The application in this tutorial _doesn't_ have the ability to send or trade [tokens](issued-currencies.html) or 
+use other [payment types](payment-types.html) like [Escrow](https://xrpl.org/escrow.html) or [Payment Channels](https://xrpl.org/payment-channels.html). However, it provides a foundation 
+that you can implement those and other features on top of.
+
+In addition to the above features, you'll also learn a bit about Events, IPC (inter-process-communication) 
+and asynchronous (async) code in JavaScript.
+
+## Steps
+
+### 0. Project setup
+
+To initialize the project we will create a package.json file with the following content:
```

Comment:
Changing the future-tense to instruct the reader to take this step.
```suggestion
To initialize the project create a package.json file with the following content:
```

Reply:
Has been rewritten